test/utils/config.go

```diff
@@ -50,6 +50,9 @@ func SetAppConfig(t *testing.T, snap, service, appName, defaultServicePort strin
 
 		const newPort = "22222"
 
+		// enable new apps option to aviod mixed options issue with old env option
+		SnapSet(t, snap, "config-enabled", "true")
+
```

Comment:
You have to unset in cleanup, otherwise the next test run will fail when checking the legacy options.

Reply:
I will unset in the cleanup. Strange, I didn't occur any failure or error without cleanup `config-enabled`: please see [logs](https://pastebin.canonical.com/p/Kfrv7mv2vV/)

Reply:
It depends on the order of execution and starting point. If snap is installed freshly and env is tested first there will be no error. Regardless, it is a good practice to cleanup what each subtest does to avoid future gotchas!